src/v2/v2_types.rs

```diff
@@ -0,0 +1,185 @@
+/// TODO: Someone remind me to doc this type.
+pub struct NodeInfo {
+    pub node_version: semver::Version,
+    pub node_type: NodeType,
+    pub local_time:   chrono::DateTime<chrono::Utc>,
+    pub node_uptime:  chrono::Duration,
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub enum BakingStatus {
+    NotInCommittee,
+    AddedButNotActiveInCommittee,
+    AddedButWrongKeys,
+    Baker(crate::types::BakerId),
+    Finalizer(crate::types::BakerId),
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub enum ConsensusStatus {
+    ConsensusNotRunning,
+    ConsensusPassive,
+    ConsensusBaking(BakingStatus),
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub struct NodeStatus {
+    pub node_info:        NodeInfo,
+    pub network_status:   NetworkStatus,
+    pub consensus_status: ConsensusStatus,
+}
+
+#[derive(Error, Debug)]
+pub enum NodeStatusError {
+    #[error("Missing NetworkStatus")]
+    MissingNetworkStatus,
+    #[error("Missing ConsensusStatus")]
+    MissingConsensusStatus,
+    #[error("Missing NodeInfo.")]
+    MissingNodeInfo,
+    #[error("Missing localtime.")]
+    MissingLocaltime,
+    #[error("Missing uptime.")]
+    MissingUptime,
+    #[error("Malformed 'Node' type. Must be either 'Bootstrapper' or 'Node'")]
+    MalformedNodeType,
+    #[error("Malformed 'CommitteeStatus")]
+    MalformedCommitteeStatus,
+    #[error("The version of the node did not conform to a semantic version: {0}")]
+    MalFormedNodeVersion(String),
+}
+
+impl From<NodeStatusError> for QueryError {
+    fn from(e: NodeStatusError) -> Self {
+        QueryError::RPCError(crate::endpoints::RPCError::ParseError(anyhow::anyhow!(e)))
+    }
+}
+
+impl TryFrom<super::generated::NodeStatus> for NodeStatus {
+    type Error = NodeStatusError;
+
+    fn try_from(node_status: super::generated::NodeStatus) -> Result<Self, Self::Error> {
+        // Parse the `NetworkStatus` and bail out if it is not present.
+        let network_status = if let Some(network_status) = node_status.network_status {
+            Ok(NetworkStatus {
+                node_id: network_status.node_id,
+                peer_total_sent: network_status.peer_total_sent,
+                peer_total_received: network_status.peer_total_received,
+                avg_bps_in: network_status.avg_bps_in,
+                avg_bps_out: network_status.avg_bps_out,
+                
+            })          
+        } else {
+            Err(NodeStatusError::MissingNetworkStatus)
+        }?;
+
+        // Parse the node type.
+        // The node type is specified via an enum:
+        // * 0 => The node is a bootstrapper and is simply relaying peers.
+        // * 1 => The node is participating in consensus either actively (baking) or passively (solely processing blocks).
+        // If the result is none of the above we return an [NodeStatusError::MalformedNodeType].
+        let node_type = match node_status.node_type {
```

Comment:
There should be a generated `from_i32` method on the type for converting enums into the proper generated rust type.